server/repository/src/migrations/v1_06_00/barcode_add_manufacturer_link_id.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    #[cfg(feature = "postgres")]
+    sql!(
+        connection,
+        r#"
+        -- Adding barcode.manufacturer_link_id
+        ALTER TABLE barcode
+        ADD COLUMN manufacturer_link_id TEXT;
+        
+        UPDATE barcode
+        SET manufacturer_link_id = manufacturer_id;
+        
+        ALTER TABLE barcode ADD CONSTRAINT barcode_manufacturer_link_id_fkey FOREIGN KEY (manufacturer_link_id) REFERENCES name_link(id);
+        "#,
+    )?;
+
+    #[cfg(not(feature = "postgres"))]
+    sql!(
+        connection,
+        r#"
+        -- Adding barcode.manufacturer_link_id
+        PRAGMA foreign_keys = OFF;
+        ALTER TABLE barcode
+        ADD COLUMN manufacturer_link_id TEXT REFERENCES name_link(id);
+        
+        UPDATE barcode
+        SET manufacturer_link_id = manufacturer_id;
+        PRAGMA foreign_keys = ON;
+     "#,
+    )?;
+
+    sql! {
+        connection,
+        r#"
+        DROP INDEX IF EXISTS index_barcode_manufacturer_id;
```

Comment:
Same as other PR


```suggestion
        DROP INDEX index_barcode_manufacturer_id;
```

Reply:
Ayyy lmao glad I ran tests rather than blind merging 
![image](https://github.com/msupply-foundation/open-msupply/assets/7684221/69d83382-2c9c-4bf0-a8e8-b5510faa7bad)